indico/htdocs/js/indico/modules/categories/categories.js

```diff
@@ -111,4 +111,96 @@
             });
         }
     };
+
+    global.setupCategoryEventList = function setupCategoryEventsList() {
+        enableIfChecked('#event-management', 'input[name=event_id]', '.js-enabled-if-checked');
+
+        function createCategoryNavigator(element, data) {
+            $('<div>').categorynavigator({
+                openInDialog: true,
+                selectLeafOnly: true,
+                onAction: function(category) {
+                    $.ajax({
+                        url: element.data('href'),
+                        type: 'POST',
+                        data: JSON.stringify($.extend({'category_id': category.id}, data || {})),
+                        dataType: 'json',
+                        contentType: 'application/json',
+                        error: handleAjaxError,
+                        success: function(data) {
+                            if (data.success) {
+                                location.reload();
+                            }
+                        }
+                    });
+                }
+            });
+        }
+
+        $('.event-management .js-move-event-to-subcategory').on('click', function(evt) {
+            createCategoryNavigator($(this));
+        });
+
+        $('.event-management-toolbar .js-move-events-to-subcategory').on('click', function(evt) {
+            var $this = $(this);
+
+            if ($this.hasClass('disabled')) {
+                return;
+            }
+
+            var data = {};
+            if ($this.data('params') && $this.data('params').all_selected) {
+                data['all_selected'] = true;
+            } else {
+                data['event_id'] = _.map($('#event-management input[name=event_id]:checkbox:checked'), function(obj) {
+                    return obj.value;
+                });
+            }
+
+            createCategoryNavigator($this, data);
+        });
+
+        function deselectRows() {
+            $('.event-management-toolbar #selection-message').hide();
+            $('.js-enabled-if-checked').data('params', {all_selected: false});
+        }
+
+        $('#event-management input[name=event_id]').on('change', function() {
+            if (!this.checked) {
+                deselectRows();
+            }
+        });
+
+        $('.event-management-toolbar [data-select-all]').on('click', function() {
```

Comment:
Should run whenever a checkbox changes its state (and then check whether all of them are selected)